tests/console/udisks2.pm

```diff
@@ -58,17 +58,26 @@ sub run {
     assert_script_run "dd if=/dev/zero of=udisk_test/testfile.data bs=512 count=10";
     assert_script_run "ls -l udisk_test/testfile.data";
     assert_script_run "mkisofs -o udisk_test.iso udisk_test/";
-    my $udloop_output = script_output("udisksctl loop-setup -r -f udisk_test.iso");
+    my $udloop_output = script_output("udisksctl loop-setup -r -f udisk_test.iso 2>&1", proceed_on_failure => 1);
 
-    die "Missing mapping info. Expected: Mapped value \nGot: /$udloop_output/"
-      unless $udloop_output =~ "Mapped";
+    if ($udloop_output =~ /^Error\ssetting\sup\sloop\sdevice.*Timed\sout\swaiting\sfor\sobject$/) {
+        record_soft_failure "boo#1177419  [Build 20201003] udisks2ctl randomly fails to set up loop device for /root/udisk_test.iso";
+    } elsif ($udloop_output =~ "Mapped") {
+        assert_script_run "losetup -j /root/udisk_test.iso | grep loop0";
```

Comment:
What's that `grep loop0` for?

Reply:
```
tw:~ # losetup -j /root/udisk_test.iso;echo $?
/dev/loop1: [0054]:275 (/root/udisk_test.iso)
/dev/loop2: [0054]:275 (/root/udisk_test.iso)
/dev/loop0: [0054]:275 (/root/udisk_test.iso)
/dev/loop3: [0054]:275 (/root/udisk_test.iso)
0
tw:~ # losetup -j /root/udisk_test.iso1;echo $?
0
```

Reply:
hm, I don't get the point, `udiskctl loop-setup` just setup a loop device, and it must not be `loop0`, or?
Or is this one of the requirements that *no* `loop*` is used before we started...
If you remove the `0` it would make more sense to me. Or grep for `(/root/udisk_test.iso)`

Reply:
yup, it should be only `loop`. I cannot guarantee that the first available loop device is exactly `loop0`.